pyserializer/validators.py

```diff
@@ -35,29 +45,43 @@ class BaseValidator(object):
     }
 
     def __init__(self, *args, **kwargs):
-        self.message = kwargs.pop(
-            'message',
-            self.default_error_messages['invalid']
-        )
+        error_messages = copy.deepcopy(self.default_error_messages)
+        error_messages.update(kwargs.pop('error_messages', {}))
+        self.error_messages = error_messages
         self.error_dict = OrderedDict([
             ('type_name', kwargs.pop('type_name', self.type_name)),
             ('type_label', kwargs.pop('type_label', self.type_label)),
-            ('message', self.message)
+            ('message', self.error_messages['invalid'])
         ])
 
     def __call__(self, value):
         # Only run the validator
         # if the value is not empty ie: (None, '', [], (), {})
         if value not in constants.EMPTY_VALUES and not self.is_valid(value):
-            raise ValidationError(
-                self.default_error_messages['invalid']
-            )
+            self.fail('invalid')
 
     def is_valid(self, value):
         raise NotImplementedError(
             '`is_valid` should be implemented by the child class.'
         )
 
+    def fail(self, key, **kwargs):
```

Comment:
Do the validators have other types of error messages other than "invalid"?

Should the error message handling happen at the field level? or does it already happen there in addition to the validators?